### PR TITLE
adding docIT-discord bot application

### PR DESCRIPTION
**Pull request by mrnigelalford**
#### Project Abstract

We have built, docIT a private Discord bot (allowing users to “talk to your docs”) that supports multiple projects or document sites, harnessing the capabilities of the latest Large Language Models (LLMs). This bot will interconnect Polkadot's key Github repositories, documentation, related articles

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Could you add the default deliverables to the [milestone tables](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality)? Also, it would help if you could add as many technical details as possible as part of the specification.

File: applications/docit-discord-docbot.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+### Development Roadmap :nut_and_bolt:
+
+### Overview
+
+- **Total Estimated Duration:** 4 weeks
+- **Full-Time Equivalent (FTE):** 3
+- **Total Costs:** 8,000 USD
+
+## Backend Development
```
Comment (mrnigelalford) [inline applications/docit-discord-docbot.md:73]: @Noc2 added additional details

File: applications/docit-discord-docbot.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+## UI Development
+App will front to Discord Bot
+Commands: Direct mention docIT with a question. DocIT will respond in a thread with response. All additional communications can exist in a threaded conversation.
+Happy to create more commands or interactions if needed.
+
+### Pre-Implementation
+
+Before diving into the milestones, we will collaborate with the Polkadot teams. Our goal is to understand their needs deeply, pinpoint areas of concern, and determine where our bot can provide the most value. The insights from these sessions will guide our choice of data sources for the following milestones.
+
+### Milestone 1 Example — Basic functionality
```
Comment (mrnigelalford) [inline applications/docit-discord-docbot.md:93]: @Noc2 added example milestone

**Comment by mrnigelalford** [comment]
> Thanks for the application. Could you add the default deliverables to the [milestone tables](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality)? Also, it would help if you could add as many technical details as possible as part of the specification.

force-updated the PR and left you comments at locations of change 🙏

**Review by takahser — CHANGES_REQUESTED** [review]
@mrnigelalford could you remove the example milestone? Also, as indicated by @Noc2 you're supposed to add the mandatory deliverables to your milestones (i.e. 0a.-0e.). Further, please add more details to the deliverables.

Regarding doing a chatbot for discord: I think this will violate our terms since each deliverable must not rely on any proprietary software to test it. Apart from that I question the value of your solution because I don't see what difference it makes whether a user would ask their questions to your bot compared to on the openai chat directly. I know that it has some limitations (e.g. trained with data not newer than 2021) but I assume that's subject to change in the future. Also, other competitors might not have this limitation (e.g. claude.ai).

**Review by semuelle — CHANGES_REQUESTED** [review]
Please provide a registered address, @mrnigelalford. If you prefer not to share it online, feel free to [email us](mailto:grants@web3.foundation?subject=[PR%201939]%20Registered%20Address).
Also, please note the comment by the CLA bot above.

**Comment by mrnigelalford** [comment]
> @mrnigelalford could you remove the example milestone? Also, as indicated by @Noc2 you're supposed to add the mandatory deliverables to your milestones (i.e. 0a.-0e.). Further, please add more details to the deliverables.
> 
> Regarding doing a chatbot for discord: I think this will violate our terms since each deliverable must not rely on any proprietary software to test it. Apart from that I question the value of your solution because I don't see what difference it makes whether a user would ask their questions to your bot compared to on the openai chat directly. I know that it has some limitations (e.g. trained with data not newer than 2021) but I assume that's subject to change in the future. Also, other competitors might not have this limitation (e.g. claude.ai).

Thanks, I'll make the application updates.
> I think this will violate our terms since each deliverable must not rely on any proprietary software to test it
Milestone testing would be done via human. Based on the state of the code-base and implementation, this request can be combined into a single milestone with human testing. 
 also, is the external LLM usage an issue for terms of service or test structure?

> I don't see what difference it makes whether a user would ask their questions to your bot compared to on the openai chat directly.
This app is a bot in discord, which means chatting in place. Also there's an assumption that increased training data in Open AI or claude.ai will provide better answers. Its possible, but inconclusive, this solution is a proven customized chat interface that returns references with every response helping

**Comment by mrnigelalford** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by mrnigelalford** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/T%26Cs.md) before we can accept your contribution. Please submit the following text as a separate comment:
> 
> I have read and hereby sign the Contributor License Agreement.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request

I have read and hereby sign the Contributor License Agreement.

**Comment by mrnigelalford** [comment]
recheck

**Comment by mrnigelalford** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by mrnigelalford** [comment]
recheck

**Comment by mrnigelalford** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by takahser — CHANGES_REQUESTED** [review]
@mrnigelalford I want to make a couple of points here:

- currently, some of the deliverables that are mandatory according to the application template are missing.
- Also, I don't understand why the milestone costs 4k but the whole grant costs 8k.
- It'd be nice to have some details on your deliverables, e.g. which technologies are you using, what is hallucination mitigation and why would it be useful?
- We don't usually support deployment efforts through our program, so feel free to remove _0e. Deployment_.
- We don't usually support other non-implementation activities like _0b. Data Crawling_, but you could limit it to the implementation of the crawling logic.

**Comment by keeganquigley** [comment]
Pinging @mrnigelalford are you still planning to submit updates? Thanks.

**Comment by semuelle** [comment]
I am closing this application due to inactivity. Please let me know if you want me to reopen it.

**Pull request closed without merging**